src/components/helpers/wrapWithApiConnect.js

```diff
@@ -1,81 +1,62 @@
 
-    componentWillUpdate(nextProps) {
-      this.loadEntities(nextProps, this.props)
-    }
+    const promiseProps = finalMapPropsToPromiseProps(rest)
+    const prevPromiseProps = promisePropsRef.current
 
-    render() {
-      const { [ELEMENT_ID_PROP_NAME]: elementId, ...rest } = this.props
-      return createElement(WrappedComponent, {
-        ...rest,
-        ...(withRef ? { ref: 'wrappedInstance' } : {}),
-      })
-    }
+    const fetchProps = Object.keys(promiseProps).reduce((props, propName) => {
+      const promiseProp = promiseProps[propName]
 
-    loadEntities(props, prevProps) {
-      const promiseProps = finalMapPropsToPromiseProps(props)
-      const prevPromiseProps = prevProps
-        ? finalMapPropsToPromiseProps(prevProps)
-        : {}
+      if (promiseProp.method !== 'fetch') {
+        return props
+      }
 
-      forEach(promiseProps, (promiseProp, propName) => {
-        const { method } = promiseProp
-
-        if (method !== 'fetch') {
-          return
-        }
+      return {
+        ...props,
+        [propName]: rest[propName]
+      }
+    }, {})
 
+    useEffect(() => {
+      Object.keys(fetchProps).forEach((propName) => {
+        const fetchProp = fetchProps[propName]
+        const promiseProp = promiseProps[propName]
         // always refresh on any change of the query or if the refresh token is set
         // and changed since the last render
         const promisePropUpdated =
           !prevPromiseProps[propName] ||
-          !promisePropsEqual(promiseProp, prevPromiseProps[propName])
+          !promisePropsEqual(promiseProps[propName], prevPromiseProps[propName])
 
-        const { fetchOnMount } = promiseProps[propName]
-        const hasJustMounted = prevProps === undefined
-        const shallForceFetch = fetchOnMount && hasJustMounted
-        const isInCache = !!props[propName].value
+        const isInCache = !!fetchProp.value
         const hasNewRefreshToken =
           promiseProp.refresh !== undefined &&
-          promiseProp.refresh !== props[propName].refresh
-        const needsFetch = shallForceFetch || !isInCache || hasNewRefreshToken
+          promiseProp.refresh !== fetchProp.refresh
+        const needsFetch = !isInCache || hasNewRefreshToken
 
         if (promisePropUpdated && needsFetch && !promiseProp.lazy) {
-          props[propName]()
+          fetchProp()
         }
       })
-    }
 
-    getWrappedInstance() {
-      invariant(
-        withRef,
-        'To access the wrapped instance, you need to specify ' +
-          '{ withRef: true } as the second argument of the connect() call.'
-      )
+      promisePropsRef.current = promiseProps
+    })
 
-      return this.refs.wrappedInstance
-    }
+    return <WrappedComponent {...rest} ref={innerRef} />
   }
 
   ApiConnect.displayName = `ApiConnect(${getDisplayName(WrappedComponent)})`
   ApiConnect.WrappedComponent = WrappedComponent
+
   return hoistStatics(ApiConnect, WrappedComponent)
```

Comment:
```suggestion
   return hoistStatics(forwardRef(ApiConnect), WrappedComponent)
```
How comes we don't need to forward refs to the parent component? is that because an ApiConnect would never use refs?

Reply:
What do you mean by "forwarding refs to the parent component"? You can only ever forward refs to children.